src/main/java/org/folio/circulation/domain/AddressTypeRepository.java

```diff
@@ -0,0 +1,31 @@
+package org.folio.circulation.domain;
+
+import static org.folio.circulation.support.Result.ofAsync;
+import static org.folio.circulation.support.Result.succeeded;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.FetchSingleRecord;
+import org.folio.circulation.support.Result;
+
+public class AddressTypeRepository {
+  private final CollectionResourceClient addressTypesStorageClient;
+
+  public AddressTypeRepository(Clients clients) {
+    addressTypesStorageClient = clients.addressTypesStorage();
+  }
+
+  public CompletableFuture<Result<AddressType>> getAddressTypeById(String id) {
```

Comment:
Has every endpoint that uses this method got the appropriate permissions for fetching address types?

Is the interface that provides this endpoint already required in the module descriptor?

Reply:
Thanks, added permission to module descriptor for check-in-by-barcode endoint, now it is the only endpoint using addressType, I guess for other endpoints permission will be added when necessary.

Interface is already required (it's "users" interface)